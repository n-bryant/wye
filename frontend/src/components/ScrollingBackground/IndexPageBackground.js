import React from "react";
import PropTypes from "prop-types";

import { withStyles } from "@material-ui/core/styles";
import createClassNameHelper from "@n_bryant/classnames-helper";
import JSS_CLASS_NAME_PREFIX from "../../../lib/classNamePrefix";

import styles from "./IndexPageBackground.styles";

/**
 * renders an animated background comprised of top multiplayer game header images
 */
export const IndexPageBackground = props => {
  const classnames = IndexPageBackground.classnames(props);

  return (
    <div className={classnames.root()}>
      <div className={classnames.element("panel")}>
        <div className={classnames.element("scrollContainer")}>
          <div className={classnames.element("scrollForward")}></div>
        </div>
        <div className={classnames.element("scrollContainer")}>
          <div className={classnames.element("scrollBackward")}></div>
        </div>
        <div className={classnames.element("scrollContainer")}>
          <div className={classnames.element("scrollForward")}></div>
        </div>
        <div className={classnames.element("scrollContainer")}>
          <div className={classnames.element("scrollBackward")}></div>
        </div>
      </div>
    </div>
  );
};
IndexPageBackground.classnames = createClassNameHelper(
  `${JSS_CLASS_NAME_PREFIX}IndexPageBackground`
);
IndexPageBackground.propTypes = {
  // styles applied to the IndexPageBackground
  classes: PropTypes.shape({
    root: PropTypes.string,
    panel: PropTypes.string,
    scrollContainer: PropTypes.string,
    scrollForward: PropTypes.string,
    scrollBackward: PropTypes.string
  })
};
IndexPageBackground.defaultProps = {
  classes: {}
};

export default withStyles(styles)(IndexPageBackground);
