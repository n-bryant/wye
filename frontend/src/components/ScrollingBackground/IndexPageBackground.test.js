import { IndexPageBackground } from "./IndexPageBackground";
import toJson from "enzyme-to-json";

describe("IndexPageBackground", () => {
  const props = {
    classes: {
      root: "root",
      panel: "panel",
      scrollContainer: "scrollContainer",
      scrollContainerStaggered: "scrollContainerStaggered",
      scrollForward: "scrollForward",
      scrollBackward: "scrollBackward"
    }
  };

  it("should render successfully", () => {
    const wrapper = shallow(<IndexPageBackground {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
