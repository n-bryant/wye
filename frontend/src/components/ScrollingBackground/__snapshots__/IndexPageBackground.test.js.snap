// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPageBackground should render successfully 1`] = `
<div
  className="wye-IndexPageBackground root"
>
  <div
    className="wye-IndexPageBackground__panel panel"
  >
    <div
      className="wye-IndexPageBackground__scrollContainer scrollContainer"
    >
      <div
        className="wye-IndexPageBackground__scrollForward scrollForward"
      />
    </div>
    <div
      className="wye-IndexPageBackground__scrollContainer wye-IndexPageBackground__scrollContainer--staggered scrollContainer scrollContainerStaggered"
    >
      <div
        className="wye-IndexPageBackground__scrollBackward scrollBackward"
      />
    </div>
    <div
      className="wye-IndexPageBackground__scrollContainer scrollContainer"
    >
      <div
        className="wye-IndexPageBackground__scrollForward scrollForward"
      />
    </div>
    <div
      className="wye-IndexPageBackground__scrollContainer wye-IndexPageBackground__scrollContainer--staggered scrollContainer scrollContainerStaggered"
    >
      <div
        className="wye-IndexPageBackground__scrollBackward scrollBackward"
      />
    </div>
  </div>
</div>
`;
